lib/componentmanager.js

```diff
@@ -429,28 +473,56 @@
      * 
      * @see ComponentManager.prototype.addComponent
      * @param {Layer} layer The layer from which to extract components
-     * @return {Array.<{errors: Array.<string>} | {component: Component}>}
+     * @return {Promise} resolves to Array.<{errors: Array.<string>} | {component: Component}>}
      */
     ComponentManager.prototype.findAllComponents = function (layer) {
```

Comment:
This might be easier to ready if it was two private functions ```_findAllComponentsFromLayerNames``` and ```_findAllComponentsFromLayerMetaData```. Then this body is just the if / else clause